### PR TITLE
Introducing `materializedViewDefinition` in SqlContainerResource

### PR DESCRIPTION
**Pull request by srwshkrshnn**
In this PR, we are introducing `materializedViewDefinition` within SqlContainerResource for creating and updating SQL Materialized View containers.

### CODE REVIEW

File: specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2022-11-15-preview/cosmos-db.json [hunk 1]
```diff
@@ -9422,6 +9426,24 @@
...
+    "MaterializedViewDefinition": {
+      "type": "object",
+      "description": "Materialized View definition for the container.",
+      "properties": {
+        "sourceCollectionId": {
+          "description": "Name of the Cosmos DB SQL source container",
+          "type": "string"
+        },
+        "definition": {
+          "description": "Definition for the SQL Materialized view",
```
Comment (chandugunturi) [inline specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2022-11-15-preview/cosmos-db.json]: Modify to "SQL query used to fetch data from source table to populate the SQL Materialized view" or in similar words. Definition may not give clear idea to end user on purpose of field.

File: specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2022-11-15-preview/examples/CosmosDBSqlMaterializedViewCreateUpdate.json [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+                          "mode": "LastWriterWins",
+                          "conflictResolutionPath": "/_ts",
+                          "conflictResolutionProcedure": ""
+                      },
+                      "geospatialConfig": {
+                          "type": "Geography"
+                      },
+                      "materializedViewDefinition": {
+                          "sourceCollectionRid": "e9N-s6Ua-4Y=",
+                          "sourceCollectionId": "source",
```
Comment (chandugunturi) [inline specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2022-11-15-preview/examples/CosmosDBSqlMaterializedViewCreateUpdate.json]: sourceContainerName is given in request body as Id.

File: specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2022-11-15-preview/cosmos-db.json [hunk 3]
```diff
@@ -9092,6 +9092,10 @@
         "createMode": {
           "description": "Enum to indicate the mode of resource creation.",
           "$ref": "#/definitions/CreateMode"
+        },
+        "materializedViewDefinition": {
+          "$ref": "#/definitions/MaterializedViewDefinition",
+          "description": "The configuration for defining Materialized View. This must be specified for creating a Materialized View container."
```
Comment (chandugunturi) [inline specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2022-11-15-preview/cosmos-db.json]: Rephrase to "This must be specified to make created container as Materialized View container."

File: specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2022-11-15-preview/examples/CosmosDBSqlMaterializedViewCreateUpdate.json [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+      "200": {
+          "body": {
+              "id": "/subscriptions/subid/resourceGroups/rg1/providers/Microsoft.DocumentDB/databaseAccounts/ddb1/sqlDatabases/databaseName/containers/mvContainerName",
+              "name": "mvContainerName",
+              "type": "Microsoft.DocumentDB/databaseAccounts/sqlDatabases/sqlContainers",
+              "location": "West US",
+              "tags": {},
+              "properties": {
+                  "resource": {
+                      "id": "mvContainer",
```
Comment (ajparule) [inline specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2022-11-15-preview/examples/CosmosDBSqlMaterializedViewCreateUpdate.json]: change to mvContainerName

File: specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2022-11-15-preview/cosmos-db.json [hunk 5]
```diff
@@ -9422,6 +9426,24 @@
         "encryptionAlgorithm"
       ]
     },
+    "MaterializedViewDefinition": {
+      "type": "object",
+      "description": "Materialized View definition for the container.",
+      "properties": {
+        "sourceCollectionId": {
+          "description": "Name of the Cosmos DB SQL source container",
```
Comment (varadahemeswari) [inline specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2022-11-15-preview/cosmos-db.json]: Name of the container on which the view needs to be created

**Review by ajparule — APPROVED** [review]

File: specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2022-11-15-preview/cosmos-db.json (hunk 3 above)
Comment (varadahemeswari) [reply specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2022-11-15-preview/cosmos-db.json]: +1. Make it clear its optional for containers unless the intention is to create a view where this field is mandatory

Comment (srwshkrshnn) [reply specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2022-11-15-preview/cosmos-db.json]: Updated the description.

File: specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2022-11-15-preview/cosmos-db.json (hunk 5 above)
Comment (srwshkrshnn) [reply specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2022-11-15-preview/cosmos-db.json]: Updated the description.

**Review by chandugunturi — APPROVED** [review]

**Pull request merged**